src/api/define.js

```diff
@@ -108,27 +108,50 @@ function generateUniqueName(name) {
   return `${name}-${rand}`;
 }
 
+function registerV0Element(uniqueName, Ctor) {
+  return document.registerElement(uniqueName, Ctor);
+}
+
+function registerV1Element(uniqueName, Ctor) {
+  window.customElements.define(uniqueName, Ctor, { extends: Ctor.extends });
+  return Ctor;
+}
+
+function prepareForRegistration(uniqueName, Ctor) {
+  Ctor[$name] = uniqueName;
+  Ctor[$props] = createInitProps(Ctor);
+  Ctor[$renderer] = createRenderer(Ctor);
+}
+
 export default function (name, opts) {
   if (opts === undefined) {
     throw new Error(`You have to define options to register a component ${name}`);
   }
-
-  const uniqueName = generateUniqueName(name);
   const Ctor = typeof opts === 'object' ? Component.extend(opts) : opts;
-
   formatLinkedAttributes(Ctor);
 
-  Ctor[$name] = uniqueName;
-  Ctor[$props] = createInitProps(Ctor);
-  Ctor[$renderer] = createRenderer(Ctor);
+  let uniqueName = generateUniqueName(name);
```

Comment:
I don't think you need this, or the checks in the local registry anymore. You can just check / try-catch against `name` and only generate a unique name if necessary, it would seem.